apis/gateway/v1beta2/apirule_types.go

```diff
@@ -65,14 +65,9 @@ type Host string
 
 // APIRuleStatus describes the observed state of ApiRule.
 type APIRuleStatus struct {
-	LastProcessedTime    *metav1.Time           `json:"lastProcessedTime,omitempty"`
-	ObservedGeneration   int64                  `json:"observedGeneration,omitempty"`
-	APIRuleStatus        *APIRuleResourceStatus `json:"APIRuleStatus,omitempty"`
-	VirtualServiceStatus *APIRuleResourceStatus `json:"virtualServiceStatus,omitempty"`
-	// +optional
-	RequestAuthenticationStatus *APIRuleResourceStatus `json:"requestAuthenticationStatus,omitempty"`
-	// +optional
-	AuthorizationPolicyStatus *APIRuleResourceStatus `json:"authorizationPolicyStatus,omitempty"`
+	LastProcessedTime  *metav1.Time           `json:"lastProcessedTime,omitempty"`
+	ObservedGeneration int64                  `json:"observedGeneration,omitempty"`
+	APIRuleStatus      *APIRuleResourceStatus `json:"APIRuleStatus,omitempty"`
```

Comment:
The APIRuleResourceStatus has a little bit different naming than we use for other resources:
```go
type APIRuleResourceStatus struct {
	Code        StatusCode `json:"code,omitempty"`
	Description string     `json:"desc,omitempty"`
}
```
Should we change `Code` to `State` to match what we have in other CRs?

Reply:
I was also wondering about moving it up and using the same thing that is used for module CRs.
Maybe this would be a better option?